pkg/agent/multicluster/stretched_networkpolicy_controller.go

```diff
@@ -0,0 +1,426 @@
+	}
+}
+
+func (s *StretchedNetworkPolicyController) processNextWorkItem() bool {
+	obj, quit := s.queue.Get()
+	if quit {
+		return false
+	}
+	defer s.queue.Done(obj)
+
+	if c, ok := obj.(*classifierFlowInfo); !ok {
+		s.queue.Forget(obj)
+		klog.Errorf("Expected type 'classifierFlowInfo' in work queue but got object", "object", obj)
+	} else if err := s.syncPodClassifierFlow(c); err == nil {
+		s.queue.Forget(c)
+	} else {
+		// Put the item back on the workqueue to handle any transient errors.
+		s.queue.AddRateLimited(c)
+		klog.ErrorS(err, "Error syncing classifierFlowInfo, requeuing", "name", c.PodRef.Name, "namespace", c.PodRef.Namespace, "labelID", c.LabelID)
+	}
+	return true
+}
+
+// syncPodClassifierFlow will update labelToPods and podToLabel and update
+// PodClassifierFlow with its LabelIdentity, if available, or UnknownLabelIdentity.
+func (s *StretchedNetworkPolicyController) syncPodClassifierFlow(classifierFlowUpdate *classifierFlowInfo) error {
+	containerConfigs := s.interfaceStore.GetContainerInterfacesByPod(classifierFlowUpdate.PodRef.Name, classifierFlowUpdate.PodRef.Namespace)
+	if len(containerConfigs) == 0 {
+		return fmt.Errorf("pod container config not found")
+	}
+	return s.ofClient.UpdatePodClassifierFlowsWithNewLabelID(
+		containerConfigs[0].InterfaceName,
+		classifierFlowUpdate.LabelID,
+		uint32(containerConfigs[0].OFPort),
+		containerConfigs[0].IPs,
+	)
+}
+
+func (s *StretchedNetworkPolicyController) getClassifierFlowUpdate(podRef types.NamespacedName, normalizedLabel string) *classifierFlowInfo {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	return s.getClassifierFlowUpdateLocked(podRef, normalizedLabel)
+}
+
+func (s *StretchedNetworkPolicyController) getClassifierFlowUpdateLocked(podRef types.NamespacedName, normalizedLabel string) *classifierFlowInfo {
+	oldNormalizedLabel, ok := s.podToLabel[podRef]
+	if ok && oldNormalizedLabel != normalizedLabel {
+		s.deleteLabelToPod(oldNormalizedLabel, podRef)
+	}
+	if !ok || oldNormalizedLabel != normalizedLabel {
+		s.addLabelToPod(normalizedLabel, podRef)
+		s.podToLabel[podRef] = normalizedLabel
+	}
+	labelID := openflow.UnknownLabelIdentity
+	if id, ok := s.labelIdentityCache[normalizedLabel]; ok {
+		labelID = id
+	}
+	return &classifierFlowInfo{
+		PodRef:  podRef,
+		LabelID: labelID,
+	}
+}
+
+func (s *StretchedNetworkPolicyController) processPodCNIAddEvent(e interface{}) {
+	podEvent := e.(antreatypes.PodUpdate)
+	if !podEvent.IsAdd {
+		return
+	}
+	podRef := types.NamespacedName{
+		Namespace: podEvent.PodNamespace,
+		Name:      podEvent.PodName,
+	}
+	pod, err := s.podLister.Pods(podRef.Namespace).Get(podRef.Name)
+	if err != nil {
+		klog.ErrorS(err, "Can't get Pod", "name", podRef.Name, "namespace", podRef.Namespace)
+		return
+	}
+	if pod.Spec.HostNetwork {
+		klog.V(2).InfoS("Received a Pod add event but decided to skip",
+			"name", pod.Name, "namespace", pod.Namespace, "node", pod.Spec.NodeName, "hostNetwork", pod.Spec.HostNetwork)
+		return
+	}
```

Comment:
It seems impossible as the event comes from CNI server.

Reply:
Good catch. Remove.